### PR TITLE
 Javascript2 week1 files added

### PR DESCRIPTION
**Pull request by Sarulathaanbu**

### CODE REVIEW

File: javascript/javascript2/week1/hyf-bay/main.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+const main = document.querySelector('main');
+const ul = document.createElement('ul')
+main.appendChild(ul)
+
+function renderProducts(products) {
+
+  for (let i = 0; i < products.length; i++) {
+    const li = document.createElement('li')
+    const name = document.createElement('h1')
```
Comment (obelitz) [inline javascript/javascript2/week1/hyf-bay/main.js:14]: Tiny thing:
I'd do `nameElement` or `nameHeader` to be more descriptive.

File: javascript/javascript2/week1/hyf-bay/main.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    const price = document.createElement('p')
+    price.innerHTML = `price: ${products[i].price}`
+    li.appendChild(price)
+
+    const rating = document.createElement('p')
+    rating.innerHTML = `rating: ${products[i].rating}`
+    li.appendChild(rating)
+
+    ul.appendChild(li)
```
Comment (obelitz) [inline javascript/javascript2/week1/hyf-bay/main.js:26]: Good job on this assignment!

File: javascript/javascript2/week1/index.html [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
+<html lang="en" dir="ltr">
+
+<head>
+  <meta charset="utf-8">
+  <title></title>
+</head>
+
+<body>
+  <script src="shortword-daletters.js"></script>
```
Comment (obelitz) [inline javascript/javascript2/week1/index.html:10]: Better a bit longer to make it more readable:
`short-words-danish-letters`

File: javascript/javascript2/week1/shortword-daletters.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+const danishWords = ["bil", "plante", "kaffe", "bog", "ø", "planetarium"];
+
+function shortWord(array) {
```
Comment (obelitz) [inline javascript/javascript2/week1/shortword-daletters.js:3]: Function names usually have verbs in the beginning:
`getShortestWord`

File: javascript/javascript2/week1/shortword-daletters.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  return array.reduce(function(prevWord, currWord) {
+    if (currWord.length < prevWord.length) {
+      return currWord;
+    } else
+      return prevWord;
```
Comment (obelitz) [inline javascript/javascript2/week1/shortword-daletters.js:8]: Nice!
You don't need the `else` here seeing as you've already returned in the if-statement

File: javascript/javascript2/week1/shortword-daletters.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+//////////////////////////Finding Danish Letters/////////////////////
+
+const danishString = "Jeg har en blå bil";
+
+function findDanishLetters(danishString) {
+  const å = danishString.match(/å/g);
+  const æ = danishString.match(/æ/g);
+  const ø = danishString.match(/ø/g);
+  let total = æ.length + å.length + ø.length;
+  return `total: ${total}, æ: ${æ.length}, ø: ${ø.length}, ${å.length}: 1`;
```
Comment (obelitz) [inline javascript/javascript2/week1/shortword-daletters.js:23]: Good thinking!

Comment (obelitz) [reply javascript/javascript2/week1/shortword-daletters.js:23]: Total should be a const

File: javascript/javascript2/week1/spirit.html [hunk 7]
```diff
@@ -0,0 +1,26 @@
+<!DOCTYPE html>
+<html lang="en" dir="ltr">
+
+<head>
+  <meta charset="utf-8">
+  <meta http-equiv="X-UA-Compatible" content="IE=edge">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <title>Document</title>
```
Comment (obelitz) [inline javascript/javascript2/week1/spirit.html:8]: Consider adding a title to these.

File: javascript/javascript2/week1/spirit.html [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+</head>
+
+<body>
+  <h1>Spirit Animal Name Generator</h1>
+  <div class="name">
+    <input type="text" , id="name">
```
Comment (obelitz) [inline javascript/javascript2/week1/spirit.html:14]: No comma needed.

File: javascript/javascript2/week1/spirit.html [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+  </div>
+
+<div class="buttons">
+  <button class="clickbutton">Click button</button>
+  <button class="hoverbutton">Hover Button</button>
+</div>
+
+<h2></h2>
+  <script src="spirit.js"></script>
```
Comment (obelitz) [inline javascript/javascript2/week1/spirit.html:23]: The indentation is off here

File: javascript/javascript2/week1/spirit.js [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+  'The Squeeky Snake',
+  'The Baby Roar',
+  'The Day Bat',
+  'The Cheeky Monkey',
+  'The Breezy Bee',
+  'The Fake Fox'
+]
+
+
+function spiritNamesGenerator(name) {
```
Comment (obelitz) [inline javascript/javascript2/week1/spirit.js:14]: `generateSpiritName`

File: javascript/javascript2/week1/spirit.js [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+  }
+
+}
+
+const clickButton = document.querySelector('.clickbutton')
+const hoverButton = document.querySelector('.hoverbutton');
+const spiritName = document.querySelector('h2');
+
+clickButton.addEventListener('click', clickButtons)
+hoverButton.addEventListener('click', mouseOver)
```
Comment (obelitz) [inline javascript/javascript2/week1/spirit.js:42]: This should probably be:
```js
hoverButton.addEventListener('mouseover', mouseOver)
```

**Review by obelitz — COMMENTED** [review]
Good job!

Notes:
- There are some naming things
- Be sure to use a code formatter (like Prettier.io) to format the code
- The mouse-over button doesn't work

File: javascript/javascript2/week1/spirit.js (hunk 11 above)
Comment (Sarulathaanbu) [reply javascript/javascript2/week1/spirit.js]: Got it!

File: javascript/javascript2/week1/spirit.js (hunk 10 above)
Comment (Sarulathaanbu) [reply javascript/javascript2/week1/spirit.js]: Sure

File: javascript/javascript2/week1/spirit.html (hunk 9 above)
Comment (Sarulathaanbu) [reply javascript/javascript2/week1/spirit.html]: Will do it.

File: javascript/javascript2/week1/spirit.html (hunk 8 above)
Comment (Sarulathaanbu) [reply javascript/javascript2/week1/spirit.html]: 👍🏻

File: javascript/javascript2/week1/spirit.html (hunk 7 above)
Comment (Sarulathaanbu) [reply javascript/javascript2/week1/spirit.html]: Sure

File: javascript/javascript2/week1/shortword-daletters.js (hunk 6 above)
Comment (Sarulathaanbu) [reply javascript/javascript2/week1/shortword-daletters.js]: 👍🏻

File: javascript/javascript2/week1/shortword-daletters.js (hunk 5 above)
Comment (Sarulathaanbu) [reply javascript/javascript2/week1/shortword-daletters.js]: 👍🏻

File: javascript/javascript2/week1/shortword-daletters.js (hunk 4 above)
Comment (Sarulathaanbu) [reply javascript/javascript2/week1/shortword-daletters.js]: 👍🏻

File: javascript/javascript2/week1/index.html (hunk 3 above)
Comment (Sarulathaanbu) [reply javascript/javascript2/week1/index.html]: Got it!

File: javascript/javascript2/week1/hyf-bay/main.js (hunk 2 above)
Comment (Sarulathaanbu) [reply javascript/javascript2/week1/hyf-bay/main.js:26]: Thank you

File: javascript/javascript2/week1/hyf-bay/main.js (hunk 1 above)
Comment (Sarulathaanbu) [reply javascript/javascript2/week1/hyf-bay/main.js]: 👍🏻

**Comment by Sarulathaanbu** [comment]
Thanks for the feedback

**Pull request merged**